src/functions.rs

```diff
@@ -2919,6 +2919,75 @@ pub fn initialize_terminated<'a>(functions: &'a FunctionMap<'a>) -> (Vec<String>
     (term_ret_vec, nonterm_ret_vec)
 }
 
+// Helper function to substitue class name if needed
+fn cil_name_helper(
+    call: FuncCall,
+    func_info: &FunctionInfo,
+    function_map: &FunctionMap,
+) -> Result<String, CascadeErrors> {
+    let mut original_cil_name = call.get_cil_name();
```

Comment:
I think it would be more idiomatic to do something like:

```
let original_cil_name = if // ... The block starting on 2931 goes here
// after 2952
else {
    call.get_cil_name()
}
```

Then you just assign it once instead of overwriting, and it won't actually need to be mutable (in order to make that happen, the second if block needs to add an else, and return both cases, rather than overwriting).